frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -579,6 +600,37 @@ common_settings.document_metadata_location = {
         },
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
+        genMetadataFolderMenuItem("hash"),
+        { -- hash-based metadata count / TextViewer
+            text_func = function()
+                local hash_text = _("Show documents with hash-based metadata")
+                local no_hash_text = _("No documents with hash-based metadata")
+                if G_reader_settings:readSetting("document_metadata_folder") == "hash" then
+                    if DocSettings.isHashLocationEnabled() then return hash_text end
+                    -- displays if the hash location setting is selected but no hash-based metadata is found
+                    return no_hash_text
+                end
+                if DocSettings.isHashLocationEnabled() then return  "⚠ "..hash_text end
+                return no_hash_text
```

Comment:
i guess the logic (to not have that many `return`) could be simplified and made clearer?
```lua
if DocSettings.isHashLocationEnabled() then
    if G_reader_settings:readSetting("document_metadata_folder") ~= "hash" then
        return  "⚠ "..hash_text
    end
    return  hash_text
end
return no_hash_text
```

Reply:
Thanks that's much clearer, changed.